main.js

```diff
@@ -447,4 +368,320 @@ function removeComparison(event) {
   }
 }
 
+var writeLeaderBoard = function updateLeaderBoard(event) {
+  showElement($leaderboard)
+  removeElement($playerStats)
+  var leaderBoardList = leaderBoardArray
+  var leaderBoardDOM = document.querySelector('#leader-board')
+  var createDiv = document.createElement('div')
+  createDiv.id = 'leader-board-rows'
+  var leaderBoardDiv = document.querySelector('#leader-board-rows')
+  if (document.querySelector('.buttons') === null) {
+    leaderBoardDOM.appendChild(leaderBoardStatButtons())
+    leaderBoardDOM.appendChild(leaderBoardMMRButtons())
+    var $buttons = document.querySelectorAll('.buttons')[0]
+    var $buttons2 = document.querySelectorAll('.buttons')[1]
+    $buttons.addEventListener('click', writeLeaderBoard)
+    $buttons2.addEventListener('click', writeLeaderBoard)
+    leaderBoardDOM.appendChild(createDiv)
+  }
+  if (event.target.textContent !== null && (event.target.className === 'statButton' || event.target.className === 'mmrButton')) {
+    clearLeaderBoard(leaderBoardDiv)
+    leaderBoardDiv.appendChild(leaderBoardHeader(event.target.textContent))
+    if (event.target.textContent === 'Solo Duel') {
+      for (var i = 0; i < 10; i++) {
+        leaderBoardDiv.appendChild(leaderBoardRow(leaderBoardList[0][i], i))
+      }
+    }
+    if (event.target.textContent === 'Doubles') {
+      for (i = 0; i < 10; i++) {
+        leaderBoardDiv.appendChild(leaderBoardRow(leaderBoardList[1][i], i))
+      }
+    }
+    if (event.target.textContent === 'Solo Standard') {
+      for (i = 0; i < 10; i++) {
+        leaderBoardDiv.appendChild(leaderBoardRow(leaderBoardList[2][i], i))
+      }
+    }
+    if (event.target.textContent === 'Standard') {
+      for (i = 0; i < 10; i++) {
+        leaderBoardDiv.appendChild(leaderBoardRow(leaderBoardList[3][i], i))
+      }
+    }
+    if (event.target.textContent === 'Wins') {
+      for (i = 0; i < 10; i++) {
+        leaderBoardDiv.appendChild(leaderBoardRow(leaderBoardList[4][i], i))
+      }
+    }
+    if (event.target.textContent === 'Goals') {
+      for (i = 0; i < 10; i++) {
+        leaderBoardDiv.appendChild(leaderBoardRow(leaderBoardList[5][i], i))
+      }
+    }
+    if (event.target.textContent === 'MVPs') {
+      for (i = 0; i < 10; i++) {
+        leaderBoardDiv.appendChild(leaderBoardRow(leaderBoardList[6][i], i))
+      }
+    }
+    if (event.target.textContent === 'Saves') {
+      for (i = 0; i < 10; i++) {
+        leaderBoardDiv.appendChild(leaderBoardRow(leaderBoardList[7][i], i))
+      }
+    }
+    if (event.target.textContent === 'Shots') {
+      for (i = 0; i < 10; i++) {
+        leaderBoardDiv.appendChild(leaderBoardRow(leaderBoardList[8][i], i))
+      }
+    }
+    if (event.target.textContent === 'Assists') {
+      for (i = 0; i < 10; i++) {
+        leaderBoardDiv.appendChild(leaderBoardRow(leaderBoardList[9][i], i))
+      }
+    }
+  }
+}
+
+function soloDuelListener() {
```

Comment:
Can you see a pattern in your implementations that could help you write some more generalized functions that alter their behavior based on arguments?